python/pyspark/sql/pandas/serializers.py

```diff
@@ -51,7 +55,20 @@ def load_stream(self, stream):
         """
         # load the batches
         for batch in self.serializer.load_stream(stream):
-            yield batch
+            if self.split_batches:
```

Comment:
This doesn't need to be done in the serializer, could you move it to `_collect_as_arrow` and leave the serializer as it was?